### ISSUE TITLE
changing tabs bug

### ISSUE THREAD

**Issue by farm-er** [godotengine/godot#95081]
### Tested versions

4.2 stable 

### System information

ubuntu 22.04.4 LTS

### Issue description

pretty straight forward: when right clicking on a scene and changing tabs the menu that appears stays on top. and it's clickable

### Steps to reproduce

create any node and open the little options menu then change tabs. you can change tabs by alt + Tab or just clicking on an icon on the side bar.

### Minimal reproduction project (MRP)

...

**Comment by farm-er** [comment]
it's version 4.2.2 stable sorry didn't specify it. And can you look at the problem I'm getting with exporting to android #95070

**Comment by MM4096** [comment]
This does happen with v4.2.2 (Ubuntu 24.04, Wayland). This also includes changing desktops with `Super + Scroll`. The panel is fully interactable, and doesn't disappear when any other window gains focus. I have also tested this out in 4.3 (rc2) and the bug is still present. See my video attached for examples.

System information: `Godot v4.3.rc2.mono - Ubuntu 24.04 LTS 24.04 - Wayland - Vulkan (Forward+) - integrated Intel(R) HD Graphics 4400 (HSW GT2) - Intel(R) Core(TM) i7-4600U CPU @ 2.10GHz (4 Threads)`

https://github.com/user-attachments/assets/fab6a2b3-bfee-48c2-89e2-7060aa122f96

**Comment by farm-er** [comment]
there is also a problem after suspending the system and getting back. Some things don't work and sometimes godot crashes after doing anything. I will investigate more and open an issue when i get the time

**Comment by chsoen** [comment]
Reproducible in v4.5.dev custom_build ([45fc515ae](https://github.com/godotengine/godot/commit/45fc515ae))
```
Godot v4.5.dev (45fc515ae) - Fedora Linux 42 (Workstation Edition) on Wayland - X11 display driver, Multi-window, 1 monitor - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1650 (NVK TU117) - AMD Ryzen 5 4600H with Radeon Graphics (12 threads)
```

**Comment by chsoen** [comment]
Not reproducible on windows 11 with Godot v4.5dev.custom_build [45fc515ae]
```
Godot v4.5.dev (45fc515ae) - Windows 11 (build 26100) - Multi-window, 1 monitor - Vulkan (Forward+) - dedicated AMD Radeon RX 6750 XT (Advanced Micro Devices, Inc.; 32.0.12033.1030) - 12th Gen Intel(R) Core(TM) i7-12700KF (20 threads)
```

**Comment by YeldhamDev** [comment]
This is true for X11. In Wayland (and with `prefer_wayland` enabled) this doesn't happen.

**Comment by BigFriendlyGeek42** [comment]
OK, so from what I could gather, only `PopupMenu`s have that behavior, not all Popups, or other Popup classes (`PopupPanel`...), this behavior is reproducible with every `PopupMenu` of the engine on Linux, with X11. I'm searching for the cause of this issue but I didn't find it yet.

**Comment by BigFriendlyGeek42** [comment]
Actually, what happens when a `PopupMenu` appears is that, whatever happens, the `NOTIFICATION_APPLICATION_FOCUS_OUT` notification will never be received (not only by the `PopupMenu`, but also by other classes, such as `Popup`s) while the aforesaid `PopupMenu` is visible, so I can assume that what happens is that the `PopupMenu` keeps focus until it is closed via input means (mouse click or ESC, etc...).

**Comment by BigFriendlyGeek42** [comment]
I tried many things with window flags (via `window_set_flag`) and I am pretty sure that this issue is not linked to flags, since the only difference between the `PopupMenu` flags and the other `Popup` flags is just that `Popup` sets the `WINDOW_FLAG_MOUSE_PASSTHROUGH` flag (and `PopupMenu` doesn't ) while the `PopupMenu` sets the `WINDOW_FLAG_POPUP` (and `Popup` doesn't ). I've tried commenting the code executing when flag is `WINDOW_FLAG_POPUP` (the only flag that is triggered by `PopupMenu` and not by `Popup`), and it doesn't change anything.
